From 4b4980792326c9f0c1fba01292319450b9b3a20d Mon Sep 17 00:00:00 2001
From: Johan Hovold <johan+linaro@kernel.org>
Date: Tue, 17 Jan 2023 11:53:44 +0100
Subject: [PATCH 144/230] fixup: firmware: qcom_tee_uefisecapp: drop the
 efivars kobject

Drop the otherwise unused kobject to match the new efivars interface.

Signed-off-by: Johan Hovold <johan+linaro@kernel.org>
---
 drivers/firmware/qcom_tee_uefisecapp.c | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/drivers/firmware/qcom_tee_uefisecapp.c b/drivers/firmware/qcom_tee_uefisecapp.c
index 65573e4b815..42894f57e54 100644
--- a/drivers/firmware/qcom_tee_uefisecapp.c
+++ b/drivers/firmware/qcom_tee_uefisecapp.c
@@ -267,7 +267,6 @@ struct qctee_rsp_uefi_query_variable_info {
 
 struct qcuefi_client {
 	struct device *dev;
-	struct kobject *kobj;
 	struct efivars efivars;
 	struct qctee_dma dma;
 	u32 app_id;
@@ -693,13 +692,6 @@ static int qcom_uefisecapp_probe(struct platform_device *pdev)
 	if (status)
 		return status;
 
-	/* Set up kobject for efivars interface. */
-	qcuefi->kobj = kobject_create_and_add("qcom_tee_uefisecapp", firmware_kobj);
-	if (!qcuefi->kobj) {
-		status = -ENOMEM;
-		goto err_kobj;
-	}
-
 	/* Register global reference. */
 	platform_set_drvdata(pdev, qcuefi);
 	status = qcuefi_set_reference(qcuefi);
@@ -707,7 +699,7 @@ static int qcom_uefisecapp_probe(struct platform_device *pdev)
 		goto err_ref;
 
 	/* Register efivar ops. */
-	status = efivars_register(&qcuefi->efivars, &qcom_efivar_ops, qcuefi->kobj);
+	status = efivars_register(&qcuefi->efivars, &qcom_efivar_ops);
 	if (status)
 		goto err_register;
 
@@ -716,8 +708,6 @@ static int qcom_uefisecapp_probe(struct platform_device *pdev)
 err_register:
 	qcuefi_set_reference(NULL);
 err_ref:
-	kobject_put(qcuefi->kobj);
-err_kobj:
 	qctee_dma_free(qcuefi->dev, &qcuefi->dma);
 	return status;
 }
@@ -733,7 +723,6 @@ static int qcom_uefisecapp_remove(struct platform_device *pdev)
 	qcuefi_set_reference(NULL);
 
 	/* Free remaining resources. */
-	kobject_put(qcuefi->kobj);
 	qctee_dma_free(qcuefi->dev, &qcuefi->dma);
 
 	return 0;
-- 
2.39.0

